### PR TITLE
change library usage for api call

### PR DESCRIPTION
**Pull request by gziebold71**

### CODE REVIEW

File: steps/api.steps.ts [hunk 1]
```diff
@@ -3,22 +3,25 @@ import {Given, When, Then} from "@cucumber/cucumber";
...
-    });
+    apiResponse = await request(config.exchangeApiUrl).get(uri).set('Accept', 'application/json')
 });
 
 Then('Verify the response status code is {string}', async function (responseCode) {
-    expect(apiResponse.ok()).toBeTruthy();
+    expect(apiResponse.statusCode).toEqual(parseInt(responseCode));
+})
+
+Then('I verify that {string} offsets are returned', async (countOfOffsets) => {
```
Comment (crespia) [inline steps/api.steps.ts:18]: can {int} be used instead of {string} and avoid the parseInt later in the code?

File: steps/api.steps.ts [hunk 2]
```diff
@@ -3,22 +3,25 @@ import {Given, When, Then} from "@cucumber/cucumber";
...
+    const body = await JSON.parse(apiResponse.text)
+    expect(body.offsets.length).toEqual(parseInt(countOfOffsets))
 })
 
 Then('I verify that all returned offsets are of type {string}', async (listState) => {
```
Comment (crespia) [inline steps/api.steps.ts:23]: I think we can remove all of the "I verify that " phrases and the scenarios will read more naturally.

File: steps/api.steps.ts [hunk 3]
```diff
@@ -3,22 +3,25 @@ import {Given, When, Then} from "@cucumber/cucumber";
...
-    const body = await apiResponse.json()
+    const body = await JSON.parse(apiResponse.text)
     for (let i = 0; i < body.offsets.length; i++) {
```
Comment (crespia) [inline steps/api.steps.ts:25]: There is a more concise notation for "for" loops where you don't need the index. something like this also works.

for (let offset of body.offsets) {
   expect(offset.list_state).toEqual(listState);
}

another technique is that the array contains a forEach() function:

body.offsets.forEach( offset =>{ expect(offset.list_state).toEqual(listState); })

File: steps/api.steps.ts (hunk 1 above)
Comment (gziebold71) [reply steps/api.steps.ts:18]: unfortunately not

File: steps/api.steps.ts (hunk 2 above)
Comment (gziebold71) [reply steps/api.steps.ts:23]: I've been using this format for a while now, so I'd honestly like to keep it.

**Review by crespia — APPROVED** [review]

**Pull request merged**